### PR TITLE
add generic JuliaBundle easyblock

### PR DESCRIPTION
**Pull request by lexming**

### CODE REVIEW

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/juliabundle.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        prev_enable_templating = self.cfg.enable_templating
+        self.cfg.enable_templating = False
+
+        # set default options for extensions according to relevant top-level easyconfig parameters
+        jlpkg_keys = JuliaPackage.extra_options().keys()
+        for key in jlpkg_keys:
+            if key not in self.cfg['exts_default_options']:
+                self.cfg['exts_default_options'][key] = self.cfg[key]
+
+        self.cfg.enable_templating = prev_enable_templating
```
[the remark this answers is not in the document]
Comment (lexming) [reply easybuild/easyblocks/generic/juliabundle.py]: indeed, using `with` is way better, thanks! :+1:
fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2830/commits/118e33bc2971731d6948ce7d5f4ffbb25e3f1917

File: easybuild/easyblocks/generic/juliabundle.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        """Custom sanity check for bundle of Julia packages"""
+        if not self.cfg['sanity_check_paths']:
+            pkg_dir = os.path.join('packages', self.name)
+
+            self.cfg['sanity_check_paths'] = {
+                'files': [],
+                'dirs': [pkg_dir],
+            }
+
+        super(JuliaBundle, self).sanity_check_step(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (lexming) [reply easybuild/easyblocks/generic/juliabundle.py]: I thought it was weird that `PythonBundle` is not using `custom_paths`, but I had no idea about the reason behind it. Fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2830/commits/bde9df58e4798440f6aa96bee571a4c38357690e

**Comment by lexming** [comment]
Updated with default definition of sources to avoid filename collisions.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/juliabundle.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.easyblocks.generic.juliapackage import EXTS_FILTER_JULIA_PACKAGES, JuliaPackage
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root
+
+
+class JuliaBundle(Bundle):
+    """
+    Bundle of modules: only generate module files, nothing to build/install
```
[the remark this answers is not in the document]
Comment (lexming) [reply easybuild/easyblocks/generic/juliabundle.py]: fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2830/commits/47314783a57dd0168f901688b2ca066bb2bc6968 :+1:

**Review by boegel — APPROVED** [review]

**Pull request merged**